syncd/FlexCounter.cpp

```diff
@@ -859,7 +859,7 @@ class CounterContext : public BaseCounterContext
             _In_ sai_stats_mode_t stats_mode)
     {
         SWSS_LOG_ENTER();
-        if (!m_supportedCounters.empty() && !always_check_supported_counters)
+        if (!m_supportedCounters.empty() && !always_check_supported_counters && !dont_clear_support_counter)
```

Comment:
should we check dont_clear_support_counter before doing m_supportedCounters.clear?